build-support/bin/check_rust_formatting.sh

```diff
@@ -42,6 +42,7 @@ if [[ "${check}" == "true" ]]; then
   cmd=("${cmd[@]}" "--check")
 fi
 
+# shellcheck disable=SC2207
```

Comment:
I originally tried fixing all of the SC2207 issues but after 3 or 4 of them, lost complete confidence that my solutions were working and became frustrated with how finicky it all was. You can see the commit history for what they would have looked like.

See https://github.com/koalaman/shellcheck/wiki/SC2207. I don't think we can use their solution of `mapfile` because `command -v mapfile` fails on macOS.

Reply:
Can we make a tracking issue and link it in the places we disable this check? Alternatively, if we don't think it's going to be worth the effort to fix at all/ever, we could perhaps note that somehow (not sure how) so nobody else falls down this rabbit hole?

Reply:
I don't think it's going to be worth the effort to fix this ever. Should we add a `NB` comment about it for each of the ~6 instances?

Reply:
Hence the (not sure how)! We could potentially add an extended comment in `shellcheck.py` listing any "WONTFIX"-esque shellchecks?

Reply:
Maybe an issue in the repo, and link to the issue from the code? Sort of a "TO_NOT_DO(#issue)"

Reply:
I will go with the style of `WONTFIX` with a one-line description that links to shellcheck's wiki. Will also add this for the disables that we added in prior PRs.